airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala

```diff
@@ -108,16 +114,17 @@ object TypeResolver extends LogSupport {
   object resolveAggregationKeys extends RewriteRule {
     def apply(context: AnalyzerContext): PlanRewriter = {
       case a @ Aggregate(child, selectItems, groupingKeys, having, _) =>
-        val resolvedChild   = resolveRelation(context, child)
-        val inputAttributes = resolvedChild.outputAttributes
+        val resolvedChild         = resolveRelation(context, child)
+        val childOutputAttributes = resolvedChild.outputAttributes
         val resolvedGroupingKeys =
           groupingKeys.map(x => {
-            val e = resolveExpression(context, x.child, inputAttributes, false)
+            val e = resolveExpression(context, x.child, childOutputAttributes, false)
             GroupingKey(e, e.nodeLocation)
           })
         val resolvedHaving = having.map {
           _.transformUpExpression { case x: Expression =>
-            resolveExpression(context, x, a.outputAttributes, false)
+            // Having recognize attributes only from the input relation
+            resolveExpression(context, x, childOutputAttributes, false)
```

Comment:
This part had a bug, which was found in 
https://github.com/wvlet/airframe/pull/2649#discussion_r1052424024